logstash_async/transport.py

```diff
@@ -176,3 +224,117 @@ def send(self, events, use_logging=False):
         with client:
             for events_subset in ichunked(events, self._batch_size):
                 client.send(events_subset)
+
+
+class HttpTransport(Transport):
+    """The :class:`HttpTransport <HttpTransport>` implements a client for the
+    logstash plugin `inputs_http`.
+
+    For more details visit:
+    https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http.html
+
+    :param host: The name of the host
+    :type host: str
+    :param port: The port number of the service
+    :type port: int
+    :param timeout: The timeout for the connection (Default: 2.0 seconds)
+    :type timeout: float
+    :param ssl_enable: Use TLS for the transport (Default: True)
+    :type ssl_enable: bool
+    :param ssl_verify: If True the class tries to verify the TLS certificate
+    with certifi. If you pass a string with a file location to CA certificate
+    the class tries to validate it against it. (Default: True)
+    :type ssl_verify: bool or str
+    :param username: Username for basic authorization (Default: "")
+    :type username: str
+    :param password: Password for basic authorization (Default: "")
+    :type password: str
+    """
+
+    def __init__(
+            self,
+            host,
+            port,
+            timeout=2.0,
+            ssl_enable=True,
+            ssl_verify=True,
+            **kwargs
+    ):
+        super().__init__(host, port, timeout, ssl_enable, ssl_verify)
+        self.username = kwargs.get('username', None)
+        self.password = kwargs.get('password', None)
+        self.__session = None
+        self.__max_attempts = 3
+
+    @property
+    def url(self):
+        """The URL of the logstash pipeline based on the hostname, the port and
+        the TLS usage.
+
+        :return: The URL of the logstash pipeline
+        :rtype: str
+        """
+        protocol = 'http'
+        if self.ssl_enable:
+            protocol = 'https'
+        return '{}://{}:{}'.format(protocol, self.host, self.port)
+
+    def encode(self, events):
```

Comment:
Could be private. But doesn't make much difference, so skip if you want.

Reply:
You are right, it should be private.